codebuild/bin/grep_simple_mistakes.sh

```diff
@@ -224,6 +224,56 @@ if [[ -n $S2N_ENSURE_WITH_INVALID_ERROR_CODE ]]; then
   printf "$S2N_ENSURE_WITH_INVALID_ERROR_CODE\n\n"
 fi
 
+#############################################
+# Assert tests don't specify the "default" security policy.
+#
+# Since the "default" policies are subject to change, tests should instead specify
+# an immutable numbered policy to avoid unwanted testing behavior.
+#############################################
+S2N_DEFAULT_SECURITY_POLICY_USAGE=$(find "$PWD" -type f -name "s2n*.c" -not -path "*/bindings/*" -not -path "*/bin/*")
+declare -A KNOWN_DEFAULT_USAGE
+KNOWN_DEFAULT_USAGE["$PWD/tls/s2n_config.c"]=1
+KNOWN_DEFAULT_USAGE["$PWD/tls/s2n_security_policies.c"]=1
```

Comment:
If the default security policy is being set somewhere in the actual code, I feel like there will probably always be a valid reason to do that. Should the search path just be limited to tests?